modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/jaxrs/exception/mapper/VirtualSettingAttachmentExceptionMapper.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.headless.commerce.admin.catalog.internal.jaxrs.exception.mapper;
+
+import com.liferay.commerce.product.type.virtual.exception.CPDefinitionVirtualSettingException;
+import com.liferay.portal.vulcan.jaxrs.exception.mapper.BaseExceptionMapper;
+import com.liferay.portal.vulcan.jaxrs.exception.mapper.Problem;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Stefano Motta
+ */
+@Component(
+	property = {
+		"osgi.jaxrs.application.select=(osgi.jaxrs.name=Liferay.Headless.Commerce.Admin.Catalog)",
+		"osgi.jaxrs.extension=true",
+		"osgi.jaxrs.name=Liferay.Headless.Commerce.Admin.Catalog.VirtualSettingAttachmentExceptionMapper"
+	},
+	service = ExceptionMapper.class
+)
+public class VirtualSettingAttachmentExceptionMapper
+	extends BaseExceptionMapper<CPDefinitionVirtualSettingException> {
+
+	@Override
+	protected Problem getProblem(
+		CPDefinitionVirtualSettingException
+			cpDefinitionVirtualSettingException) {
+
+		return new Problem(
+			Response.Status.BAD_REQUEST, "Missing virtual attachment and url");
```

Comment:
Let's change it to "Invalid product virtual settings."

Reply:
Done